File: uniffi/src/message/groups/non_trans_receipt_couples.rs
use cesrox::error::ParsideResult;
use cesrox::Group;
use cesrox::Cigar;
pub use cesrox::message::groups::{
    NonTransReceiptCouples,
    NonTransReceiptCouple,
};

pub fn non_trans_receipt_couple_create(cigar: Cigar) -> NonTransReceiptCouple {
    NonTransReceiptCouple::new(cigar)
}

pub fn non_trans_receipt_couples_create(value: Vec<NonTransReceiptCouple>) -> NonTransReceiptCouples {
    NonTransReceiptCouples::new(value)
}

pub fn non_trans_receipt_couples_qb64(non_trans_receipt_couples: &NonTransReceiptCouples) -> ParsideResult<String> {
    non_trans_receipt_couples.qb64()
}

pub fn non_trans_receipt_couples_qb64b(non_trans_receipt_couples: &NonTransReceiptCouples) -> ParsideResult<Vec<u8>> {
    non_trans_receipt_couples.qb64b()
}

pub fn non_trans_receipt_couples_qb2(non_trans_receipt_couples: &NonTransReceiptCouples) -> ParsideResult<Vec<u8>> {
    non_trans_receipt_couples.qb2()
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/groups/NonTransReceiptCouples.kt
package cesrox.groups

import cesrox.primitives.Cigar
import cesrox.primitives.Verfer
import uniffi.cesrox.*
import uniffi.cesrox.NonTransReceiptCouple as NonTransReceiptCoupleNative
import uniffi.cesrox.NonTransReceiptCouples as NonTransReceiptCouplesNative

class NonTransReceiptCouple {
    val value: NonTransReceiptCoupleNative

    constructor(cigar: Cigar) {
        value = nonTransReceiptCoupleCreate(cigar.value)
    }

    constructor(nativeValue: NonTransReceiptCoupleNative) {
        value = nativeValue
    }

    val cigar get() = Cigar(value.cigar)
    val verfer get() = Verfer(value.cigar.verfer)
}

class NonTransReceiptCouples {
    val value: NonTransReceiptCouplesNative

    constructor(couples: List<NonTransReceiptCouple>) {
        value = nonTransReceiptCouplesCreate(couples.map { it.value })
    }

    constructor(couples: NonTransReceiptCouplesNative) {
        value = couples
    }

    fun get(index: Int): NonTransReceiptCouple {
        return NonTransReceiptCouple(value.value[index])
    }

    fun qb2(): List<UByte> {
        return nonTransReceiptCouplesQb2(value)
    }
    fun qb64b(): List<UByte> {
        return nonTransReceiptCouplesQb64b(value)
    }
    fun qb64(): String {
        return nonTransReceiptCouplesQb64(value)
    }
}